_posts/2020-04-01-Robust Apps for the cloud.md

```diff
@@ -0,0 +1,253 @@
+
+1. In the Performance pane click on "Configure Profiler"
+
+    ![ai-1]({{site.baseurl}}/media/2020/04/aiprofiler-1.jpg)
+
+1. In the pane that opens after that, click on "Profile Now" to start profiling.
+
+    ![ai-2]({{site.baseurl}}/media/2020/04/aiprofiler-2.jpg)
+
+1. When Profiler is running, it profiles randomly about once per hour and for a duration of two minutes. If your application is handling a steady stream of requests, Profiler uploads traces every hour. 
+To view traces, in the Performance pane, select Take Actions, and then select the Profiler Traces button. 
+
+    ![ai-3]({{site.baseurl}}/media/2020/04/aiprofiler-3.jpg)
+
+1. App Insights also allows you to track dependencies in your application. You can leverage this feature to troubleshoot slow requests. To automatically track dependencies from .NET console apps, install the Nuget package `Microsoft.ApplicationInsights.DependencyCollector`, and initialize `DependencyTrackingTelemetryModule` as follows:
+
+    ```c#
+    DependencyTrackingTelemetryModule depModule = new DependencyTrackingTelemetryModule();
+    depModule.Initialize(TelemetryConfiguration.Active);
+    ```
+
+1. Each request event is associated with the dependency calls, exceptions, and other events that are tracked while your app is processing the request. So if some requests are doing badly, you can find out whether it's because of slow responses from a dependency. You can see a waterfall view of the requests in the performance blade as well under the "Dependencies" tab:
+
+    ![ai-4]({{site.baseurl}}/media/2020/04/dependency.jpg)
+
+You can also leverage our newly released [App Insights integration with App Service Diagnostics](https://azure.github.io/AppService/2020/04/21/Announcing-Application-Insights-Integration-with-App-Service-Diagnostics.html), discussed in details here:
+
+**Learn More**
+
+- [Profile production applications in Azure with Application Insights](https://docs.microsoft.com/azure/azure-monitor/app/profiler-overview)
+- [Diagnose exceptions in your web apps with Application Insights](https://docs.microsoft.com/azure/azure-monitor/app/asp-net-exceptions)
+- [Dependency Tracking in Azure Application Insights](https://docs.microsoft.com/azure/azure-monitor/app/asp-net-dependencies#diagnosis)
+
+## Deploy in Multiple Regions
+
+In the event that a catastrophic incident happens in one of the Azure Datacenters, you can still guarantee that your app will run and serve requests by investing in Azure Front Door or Traffic Manager. There are additional benefits to using Front Door or Traffic Manager, such as routing incoming requests based the customers' geography to provide the shortest respond time to customers and distribute the load among your instances.
```

Comment:
I would reorganize this to first explain that AFD is deployed in front of the sites and distributes traffic. Right now the opening sentence is not clear how AFD protects me if there's an incident.

Reply:
Done :)